// vite-plugin-radar/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  VitePluginRadar: () => VitePluginRadar,
  default: () => src_default
});
module.exports = __toCommonJS(src_exports);

// src/google-analytics.ts
var GTagSource = "https://www.googletagmanager.com";
var GTagBase = (source) => `${source}/gtag/js`;
function injectTag(options) {
  const tags = [];
  let properties = [];
  if (Array.isArray(options)) {
    properties.push(
      ...options
    );
  } else {
    properties.push(options);
  }
  properties = properties.filter((property) => Boolean(property.id));
  if (!properties.length)
    return tags;
  const mainProperty = properties.shift();
  if (!mainProperty)
    return tags;
  let template = "";
  if (mainProperty.disable)
    template += `window['ga-disable-${mainProperty.id}'] = true;
`;
  for (const property of properties) {
    if (property.disable)
      template += `window['ga-disable-${property.id}'] = true;
`;
  }
  template += "window.dataLayer = window.dataLayer || [];\n";
  template += "function gtag(){dataLayer.push(arguments);}\n";
  template += "gtag('js', new Date());\n";
  for (const property of [mainProperty, ...properties]) {
    if (property.config || property.persistentValues) {
      const config = Object.assign(
        {},
        property.persistentValues,
        property.config
      );
      template += `gtag('config', '${property.id}', ${JSON.stringify(config)});
`;
    } else {
      template += `gtag('config', '${property.id}');
`;
    }
  }
  tags.push({
    tag: "script",
    attrs: {
      src: `${GTagBase((mainProperty == null ? void 0 : mainProperty.source) || GTagSource)}?id=${mainProperty.id}`,
      async: true
    }
  });
  tags.push({
    tag: "script",
    children: template
  });
  return tags;
}
var google_analytics_default = injectTag;

// src/google-tag-manager.ts
var GTMBase = "https://www.googletagmanager.com/gtm.js";
var NSBase = "https://www.googletagmanager.com/ns.html";
function injectTag2(options) {
  const tags = [];
  let properties = [];
  if (Array.isArray(options)) {
    properties.push(
      ...options
    );
  } else {
    properties.push(options);
  }
  properties = properties.filter((property) => Boolean(property.id));
  if (!properties.length)
    return tags;
  let template = "";
  template += "window.dataLayer = window.dataLayer || [];\n";
  template += "window.dataLayer.push({";
  template += "'gtm.start': new Date().getTime(),";
  template += "event:'gtm.js'";
  template += "});\n";
  tags.push({
    tag: "script",
    children: template
  });
  for (const property of properties) {
    tags.push({
      tag: "script",
      attrs: {
        src: `${GTMBase}?id=${property.id}`,
        async: true
      }
    });
    tags.push({
      tag: "noscript",
      injectTo: "body-prepend",
      children: `<iframe src="${NSBase}?id=${property.id}" height="0" width="0" style="display:none;visibility:hidden"></iframe>`
    });
  }
  return tags;
}
var google_tag_manager_default = injectTag2;

// src/facebook-pixel.ts
var PixelBase = "https://connect.facebook.net/en_US/fbevents.js";
var NoScriptBase = "https://www.facebook.com/tr";
function injectTag3(options) {
  const tags = [];
  let properties = [];
  if (Array.isArray(options)) {
    properties.push(
      ...options
    );
  } else {
    properties.push(options);
  }
  properties = properties.filter((property) => Boolean(property.id));
  if (!properties.length)
    return tags;
  let template = "";
  let noscriptTemplate = "";
  template += "!function(f,b,e,v,n,t,s)";
  template += "{if(f.fbq)return;n=f.fbq=function(){n.callMethod?";
  template += "n.callMethod.apply(n,arguments):n.queue.push(arguments)};";
  template += "if(!f._fbq)f._fbq=n;n.push=n;n.loaded=!0;n.version='2.0';";
  template += "n.queue=[];t=b.createElement(e);t.async=!0;";
  template += "t.src=v;s=b.getElementsByTagName(e)[0];";
  template += "s.parentNode.insertBefore(t,s)}(window, document,'script',";
  template += `'${PixelBase}');`;
  for (const property of properties) {
    template += `fbq('init', '${property.id}');`;
    noscriptTemplate += `<img height="1" width="1" style="display:none" src="${NoScriptBase}?id=${property.id}&ev=PageView&noscript=1"/>
`;
  }
  template += "fbq('track', 'PageView');";
  tags.push({
    tag: "script",
    children: template
  });
  tags.push({
    tag: "noscript",
    injectTo: "body-prepend",
    children: noscriptTemplate
  });
  return tags;
}
var facebook_pixel_default = injectTag3;

// src/vk-retargeting.ts
var RetargetingBase = "https://vk.com/js/api/openapi.js?169";
var NoScriptBase2 = "https://vk.com/rtrg";
function injectTag4(options) {
  const tags = [];
  let properties = [];
  if (Array.isArray(options)) {
    properties.push(
      ...options
    );
  } else {
    properties.push(options);
  }
  properties = properties.filter((property) => Boolean(property.id));
  if (!properties.length)
    return tags;
  let template = "";
  let noscriptTemplate = "";
  template += "!(function()";
  template += '{var t=document.createElement("script");t.type="text/javascript",';
  template += `t.async=!0,t.src="${RetargetingBase}",t.onload=function(){`;
  for (const property of properties) {
    template += `VK.Retargeting.Init("${property.id}"),`;
    noscriptTemplate += `<img height="1" width="1" style="display:none" src="${NoScriptBase2}?p=${property.id}"/>
`;
  }
  template += "VK.Retargeting.Hit()},document.head.appendChild(t);})();";
  tags.push({
    tag: "script",
    children: template
  });
  tags.push({
    tag: "noscript",
    injectTo: "body-prepend",
    children: noscriptTemplate
  });
  return tags;
}
var vk_retargeting_default = injectTag4;

// src/linkedin-insight.ts
var InsightBase = "https://snap.licdn.com/li.lms-analytics/insight.min.js";
var NoScriptBase3 = "https://px.ads.linkedin.com/collect/";
function injectTag5(options) {
  const tags = [];
  let properties = [];
  if (Array.isArray(options)) {
    properties.push(
      ...options
    );
  } else {
    properties.push(options);
  }
  properties = properties.filter((property) => Boolean(property.id));
  if (!properties.length)
    return tags;
  let template = "";
  let noscriptTemplate = "";
  template += "window._linkedin_data_partner_ids = window._linkedin_data_partner_ids || [];\n";
  for (const property of properties) {
    template += `window._linkedin_data_partner_ids.push(${property.id});
`;
    noscriptTemplate += `<img height="1" width="1" style="display:none;" alt="" src="${NoScriptBase3}?pid=${property.id}&fmt=gif" />`;
  }
  tags.push({
    tag: "script",
    children: template
  });
  tags.push({
    tag: "script",
    attrs: {
      src: `${InsightBase}`,
      async: true
    }
  });
  tags.push({
    tag: "noscript",
    injectTo: "body-prepend",
    children: noscriptTemplate
  });
  return tags;
}
var linkedin_insight_default = injectTag5;

// src/baidu-tongji.ts
var TongjiBase = "https://hm.baidu.com/hm.js";
function injectTag6(options) {
  const tags = [];
  let properties = [];
  if (Array.isArray(options)) {
    properties.push(
      ...options
    );
  } else {
    properties.push(options);
  }
  properties = properties.filter((property) => Boolean(property.id));
  if (!properties.length)
    return tags;
  let template = "";
  template += "var _hmt = _hmt || [];\n";
  template += "_hmt.push(['_setAutoPageview', true]);\n";
  for (const property of properties)
    template += `_hmt.push(['_setAccount', '${property.id}']);
`;
  tags.push({
    tag: "script",
    children: template
  });
  for (const property of properties) {
    tags.push({
      tag: "script",
      attrs: {
        src: `${TongjiBase}?${property.id}`,
        async: true
      }
    });
  }
  return tags;
}
var baidu_tongji_default = injectTag6;

// src/yandex-metrica.ts
var MetricaBase = "https://cdn.jsdelivr.net/npm/yandex-metrica-watch/tag.js";
var NoScriptBase4 = "https://mc.yandex.ru/watch/";
function injectTag7(options) {
  const tags = [];
  let properties = [];
  if (Array.isArray(options)) {
    properties.push(
      ...options
    );
  } else {
    properties.push(options);
  }
  properties = properties.filter((property) => Boolean(property.id));
  if (!properties.length)
    return tags;
  let template = "";
  let noscriptTemplate = "";
  template += "(function(m,e,t,r,i,k,a){m[i]=m[i]||function(){(m[i].a=m[i].a||[]).push(arguments)};";
  template += "m[i].l=1*new Date();k=e.createElement(t),a=e.getElementsByTagName(t)[0],k.async=1,k.src=r,a.parentNode.insertBefore(k,a)})";
  template += `(window, document, "script", "${MetricaBase}", "ym");`;
  for (const property of properties) {
    if (property.config)
      template += `ym(${property.id}, "init", ${JSON.stringify(property.config)});
`;
    else
      template += `ym(${property.id}, "init");
`;
    noscriptTemplate += `<img src="${NoScriptBase4}${property.id}" style="position:absolute;left:-9999px;" alt="" />`;
  }
  tags.push({
    tag: "script",
    children: template
  });
  tags.push({
    tag: "noscript",
    injectTo: "body-prepend",
    children: `<div>${noscriptTemplate}</div>`
  });
  return tags;
}
var yandex_metrica_default = injectTag7;

// src/microsoft-advertising.ts
var UetBase = "//bat.bing.com/bat.js";
function injectTag8(property) {
  const tags = [];
  if (!property.id)
    return tags;
  let template = "";
  template += "(function (w, d, t, r, u) {";
  template += `var f, n, i; w[u] = w[u] || [], f = function () { var o = { ti: "${property.id}" };`;
  template += 'o.q = w[u], w[u] = new UET(o), w[u].push("pageLoad") }, n = d.createElement(t), ';
  template += "n.src = r, n.async = 1, n.onload = n.onreadystatechange = function () { ";
  template += 'var s = this.readyState; s && s !== "loaded" && s !== "complete" || (f(), ';
  template += "n.onload = n.onreadystatechange = null) }, i = d.getElementsByTagName(t)[0], ";
  template += "i.parentNode.insertBefore(n, i) })";
  template += `(window, document, "script", "${UetBase}", "uetq");`;
  tags.push({
    tag: "script",
    children: template
  });
  return tags;
}
var microsoft_advertising_default = injectTag8;

// src/index.ts
function VitePluginRadar({
  enableDev = false,
  analytics,
  gtm,
  pixel,
  linkedin,
  tongji,
  metrica,
  microsoft,
  retargeting
}) {
  let viteConfig;
  return {
    name: "vite-plugin-Radar",
    configResolved(resolvedConfig) {
      viteConfig = resolvedConfig;
    },
    transformIndexHtml() {
      const tags = [];
      if (viteConfig.command === "serve" && !enableDev)
        return tags;
      if (analytics)
        tags.push(...google_analytics_default(analytics));
      if (gtm)
        tags.push(...google_tag_manager_default(gtm));
      if (pixel)
        tags.push(...facebook_pixel_default(pixel));
      if (retargeting)
        tags.push(...vk_retargeting_default(retargeting));
      if (tongji)
        tags.push(...baidu_tongji_default(tongji));
      if (linkedin)
        tags.push(...linkedin_insight_default(linkedin));
      if (metrica)
        tags.push(...yandex_metrica_default(metrica));
      if (microsoft)
        tags.push(...microsoft_advertising_default(microsoft));
      return tags;
    }
  };
}
var src_default = VitePluginRadar;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  VitePluginRadar
});
